addons/sale_stock/models/sale_order.py

```diff
@@ -42,6 +47,28 @@ def _default_warehouse_id(self):
     json_popover = fields.Char('JSON data for the popover widget', compute='_compute_json_popover')
     show_json_popover = fields.Boolean('Has late picking', compute='_compute_json_popover')
 
+    def _init_column(self, column_name):
+        """ Ensure the default warehouse_id is computed without considering the company_dependent field on res_users
+
+        At column initialization, the ir.model.fields for res.users.property_warehouse_id isn't created,
+        which means trying to read the field to get the default value will crash.
+        We therefore enforce the default here, without going through
+        the default function on the warehouse_id field.
+        """
+        if column_name != "warehouse_id":
+            super(SaleOrder, self)._init_column(column_name)
```

Comment:
you can remove one indentation level below with an early return:
```suggestion
            return super()._init_column(column_name)
```